python/cudf/cudf/core/column/categorical.py

```diff
@@ -812,37 +862,43 @@ def set_base_data(self, value):
         else:
             super().set_base_data(value)
 
-    def set_base_mask(self, value):
+    def set_base_mask(self, value: Optional[Buffer]):
         super().set_base_mask(value)
         self._codes = None
 
-    def set_base_children(self, value):
+    def set_base_children(self, value: Tuple["ColumnBase", ...]):
         super().set_base_children(value)
         self._codes = None
 
     @property
-    def children(self):
+    def children(self) -> Tuple["NumericalColumn"]:
```

Comment:
I don't think all children are necessarily `NumericalColumn`, only the codes.

Reply:
Categories live in dtype for now, not children. This dictates a tuple with a single NumericalColumn which is correct.